test/fpm_test/test_registry.f90

```diff
@@ -0,0 +1,81 @@
+module test_registry
+   use testsuite, only: new_unittest, unittest_t, error_t, test_failed
+   use fpm_command_line, only: fpm_registry_settings
+   use fpm_registry, only: get_registry_settings
+   use fpm_filesystem, only: is_dir, join_path, mkdir, filewrite, os_delete_dir, exists
+   use fpm_environment, only: os_is_unix
+
+   implicit none
+   private
+   public collect_registry
+
+   character(len=*), parameter :: tmp_folder = 'tmp'
+   character(len=*), parameter :: config_file_name = 'config.toml'
+
+contains
+
+   !> Collect unit tests.
+   subroutine collect_registry(tests)
+
+      !> Unit tests to collect.
+      type(unittest_t), allocatable, intent(out) :: tests(:)
+
+      tests = [ &
+      & new_unittest('no-tmp-folder', no_tmp_folder), &
```

Comment:
It's somewhat better practice to have tests not depend on each other or depend on order of execution. I would recommend just handling this as part of the other tests. I.e. just put at the beginning of each test

```fortran
if (is_dir(tmp_folder)) call os_delete_dir(os_is_unix(), tmp_folder)
```